Add CentOS 7 support to hoist

SpamapS opened the pull request.

SpamapS committed:
more ssh fixes for redhat

SpamapS committed:
adapt elasticsearch for redhat

SpamapS committed:
Adapt logstash for redhat

SpamapS committed:
Fix mysql for redhat

SpamapS committed:
port zookeeper role to centos 7

SpamapS committed:
Fixing python-app for redhat

SpamapS committed:
pip fixes for python3

SpamapS committed:
Fix nginx to work with redhat

SpamapS committed:
fix uwsgi for redhat

SpamapS committed:
Add a source builder for python3

SpamapS committed:
Make python3 work on redhat with source build

SpamapS committed:
fix zuul-executor for redhat

SpamapS committed:
fix logrotate for redhat

SpamapS committed:
Fix filebeat for redhat

SpamapS committed:
Fix fail2ban for redhat and python3-based ansible

SpamapS committed:
Fix integration-handler for redhat

SpamapS committed:
Fix nodepool for redhat

SpamapS committed:
fix monitoring for nginx

SpamapS committed:
Make it possible to disable datadog by var

SpamapS committed:
hack to get bwrap 0.1.8 on centos

jamielennox commented on roles/common/tasks/main.yml:
```diff
@@ -3,12 +3,23 @@
     - net-tools
 
+- name: Install common packages
+  yum:
```
when: ansible_os_family == "Redhat" ??

jamielennox replied:
the other way to do all this in include_vars: "{{ ansible_os_family }" and use package: instead of apt/yum. Whatever.

SpamapS replied:
Oooops forgot the when! :) As far as using 'package:' ... at what point does it become unreadable and you have to dig in 3 different files to find the list of packages?

jamielennox commented on roles/apache/tasks/main.yml:
```diff
@@ -1,4 +1,8 @@
 ---
 - include: apache.yml
+  when: ansible_os_family == "Debian"
```
nit: this seems reasonable, but if we're going to do apache_redhat can we rename this apache_ubuntu

SpamapS replied:
I'll rename to redhat.yaml to match the later patches (as you can tell this was an earlier patch. Also later I abandoned apache and used nginx, so I'm not 100% sure this even works.. I'm also not sure why we have apache still)

jamielennox commented on roles/apache/tasks/ssl_redhat.yml:
```diff
@@ -0,0 +1,9 @@
+    mode: 0644
+    owner: root
+    group: root
+  notify: restart apache
```
feels like this could be merged into existing with some vars, but i'm not going to get picky with all this.

SpamapS replied:
I like that approach too. This is lazy, but it does work.

jamielennox commented on roles/sshd/tasks/main.yml:
```diff
@@ -8,6 +8,6 @@
 - name: ensure ssh started
   service:
-    name: ssh
+    name: "{{ 'ssh' if ansible_os_family == 'Debian' else 'sshd' }}"
```
nit: not a blocker, but yea i would prefer to do include_var stuff rather than ifs inline

SpamapS replied:
When we add a third distro, we should totally do that. ;)

jamielennox commented on roles/apache/tasks/apache_redhat.yml:
```diff
@@ -13,6 +13,12 @@
+
 - meta: flush_handlers
 
 - name: start apache
```
squash into first commit?

SpamapS replied:
Done

jamielennox commented on roles/mysql/tasks/debian.yml:
```diff
@@ -0,0 +1,42 @@
+   set_fact: mysql_flavor=mysql
+
+ - name: Set mysql_daemon_name
+   set_fact: mysql_daemon_name=mysqld
```
vars instead of set_fact?

SpamapS replied:
I don't actually know how to do that. An example of how you'd like it done would be helpful, thanks.

jamielennox commented on roles/zookeeper/tasks/upstream_install.yml:
```diff
@@ -0,0 +1,34 @@
+  unarchive:
+    src: /usr/local/zk/zookeeper-3.4.10.tar.gz
+    dest: /usr/local
+    remote_src: yes
```
is this idempotent? personal opinion, i'd try and find an rpm.

SpamapS replied:
It is because it is version specific. RPMs would require repos, and I don't have one that is reliable. You got one? I'll fix this happily!

jamielennox commented on roles/pip/vars/main.yml:
```diff
@@ -1,2 +1,3 @@
 ---
 pip_python_exec: "{{ pip_python3 | bool | ternary('python3', 'python') }}"
+pip_exec: "{{ pip_python3 | bool | ternary('pip3', 'pip') }}"
```
ok, surprised that this isn't handled by the python that you installed it wiht.

SpamapS replied:
It's because of the altinstall. We keep the system level pip executable.

jamielennox commented on roles/sshd/handlers/main.yml:
```diff
@@ -1,4 +1,4 @@
   service:
-    name: ssh
+    name: "{{ 'ssh' if ansible_os_family == 'Debian' else 'sshd' }}"
     state: restarted
```
squash,

SpamapS replied:
Done

jamielennox commented on roles/python-app/vars/main.yml:
```diff
@@ -1,2 +1,3 @@
 ---
 python_app_python_exec: "{{ python_app_python3 | bool | ternary('python3', 'python') }}"
+python_app_pip_exec: "{{ pip_python3 | bool | ternary('pip3', 'pip') }}"
```
squash?

SpamapS replied:
Done

jamielennox commented on roles/nginx/tasks/redhat.yml:
```diff
@@ -0,0 +1,40 @@
+- name: Install nginx
+  yum:
+    name:
+      - nginx
```
nit: this looks off. with_items?

SpamapS replied:
Yum works better if you pass all the packages in at once. Apt doesn't allow this, but yum does.

jamielennox commented on roles/python-app/vars/main.yml:
```diff
@@ -1,3 +1,3 @@
-python_app_python_exec: "{{ python_app_python3 | bool | ternary('python3', 'python') }}"
-python_app_pip_exec: "{{ pip_python3 | bool | ternary('pip3', 'pip') }}"
+python_app_python_exec: "{{ python_app_python3 | bool | ternary('python3.5', 'python') }}"
+python_app_pip_exec: "{{ pip_python3 | bool | ternary('pip3.5', 'pip') }}"
```
feels like this would cause us upgrade problems.

SpamapS replied:
It will only cause problems when you go to 3.6. A bridge to cross later?

jamielennox commented on roles/zuul-executor/tasks/main.yml:
```diff
@@ -30,18 +30,26 @@
+  when: ansible_os_family == 'RedHat'
+  yum:
+    name:
+      - bubblewrap
```
does this even work? with_items or name: bubblewrap right?

jamielennox replied:
can just use package: for this because same name across dists.

SpamapS replied:
TIL about package:

jamielennox commented on roles/logrotate/tasks/main.yml:
```diff
@@ -1,9 +1,15 @@
+- name: Install base package
+  when: ansible_os_family == 'RedHat'
+  yum:
+    name: logrotate
```
package:

jamielennox commented on roles/filebeat/tasks/main.yml:
```diff
@@ -1,9 +1,15 @@
+- name: Install Fielbeat.
+  when: ansible_os_family == 'RedHat'
+  yum:
+    name: filebeat
```
package:

jamielennox commented on roles/integration-handler/handlers/main.yml:
```diff
@@ -1,7 +1,7 @@
 - name: Reload integration-handler uwsgi
   file:
-    path: /etc/uwsgi-emperor/vassals/integration-handler.ini
+    path: "{{ '/etc/uwsgi-emperor/vassals' if ansible_os_family == 'Debian' else '/etc/uwsgi.d' }}/integration-handler.ini"
```
it seems weird to me that this isn't handled by the role dependency

jamielennox commented on roles/nodepool/tasks/main.yml:
```diff
@@ -93,8 +106,22 @@
-      - "--chown=nodepool:nodepool"
+# The option below is amazing, but requires rsync 3.1+
+#    rsync_opts:
+#      - "--chown=nodepool:nodepool"
```
we should look at pulling these out into their own git repo

SpamapS replied:
Agreed. This solution, btw, SUCKS. It changes things every time. I'd much rather use --chown. I might even be willing to chase down upgrade rsync for centos 7. But.. this was the quickest way I got to a working nodepool on a short schedule. Open to other suggestions too.

jamielennox commented on roles/nodepool/tasks/main.yml:
```diff
@@ -93,8 +106,22 @@
+
+- name: Get list of files in /etc/nodepool
+  command: find /etc/nodepool -type f \! -name nodepool.yaml -a \! -name secure.conf
+  register: files_in_etc_nodepool
```
why aren't nodepool.yaml and secure.conf owned by nodepool anyway?

jamielennox replied:
Feels like there is a better way we could organize all this stuff, maybe put nodepool templates in opt?

jamielennox commented on roles/zookeeper/files/usr/lib/systemd/system/zookeeper.service:
```diff
@@ -5,9 +5,11 @@ After=syslog.target
+Group=zookeeper
+WorkingDirectory=/var/lib/zookeeper
+Environment="ZOOCFGDIR=/etc/zookeeper/conf"
+ExecStart=/usr/share/zookeeper/bin/zkServer.sh start-foreground
```
squash

SpamapS replied:
Done

SpamapS committed:
ssh fixes for redhat

We don't need to bother installing sshd. Anywhere ansible can reach
is likely to have it.

SpamapS committed:
Update apache role to support redhat

SpamapS committed:
Update common role to support redhat

SpamapS commented:
FYI, I did some squashes so it's at least a few less patches. No less daunting as a review task though. Sorry for that.
